controllers/snapshot/snapshot_adapter.go

```diff
@@ -306,6 +306,17 @@ func (a *Adapter) EnsureCreationOfEphemeralEnvironments() (controller.OperationR
 					"integrationTestScenario.Name", integrationTestScenario.Name)
 				return controller.StopProcessing()
 			}
+			if clienterrors.IsInvalid(err) {
+				a.logger.Error(err, "environment or deploymentTargetClass referenced by integrationScenario is invalid", "integrationTestScenario.Name", integrationTestScenario.Name)
+				if !gitops.IsSnapshotMarkedAsFailed(a.snapshot) {
+					_, err = gitops.MarkSnapshotAsFailed(a.client, a.context, a.snapshot,
+						fmt.Sprintf("The environment or deploymentTargetClass associated with snapshot %s is invalid.", a.snapshot.Spec.Application))
```

Comment:
Should we also propagate the error also here, to show users why it's invalid?

Reply:
`associated with snapshot` it's not associated by snapshot, but by integration test scenario (at least environment)

Reply:
Error propagation and correct info on relation to snapshot vs ITS in [b674ec](https://github.com/redhat-appstudio/integration-service/pull/421/commits/b674ecd242f54c9700dd14f99c04946a91f6af72)